[PyTorch][Edge] Reuse readArchiveAndTensors in mobile import

cccclai opened the pull request:
## Summary 

The function from jit `torch::jit::readArchiveAndTensors` is quite general and can be reused in `mobile/import.h`. 

## Test plan
CI

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56996 [PyTorch][Edge] Reuse readArchiveAndTensors in mobile import**

cccclai committed:
[PyTorch][Edge] Reuse readArchiveAndTensors in mobile import

[ghstack-poisoned]

iseeyuan commented on torch/csrc/jit/mobile/import.cpp:
```diff
@@ -6,8 +6,8 @@
 #include <torch/csrc/jit/mobile/interpreter.h>
 #include <torch/csrc/jit/mobile/observer.h>
 #include <torch/csrc/jit/runtime/instruction.h>
+#include <torch/csrc/jit/serialization/import.h>
```
The definition of `readArchiveAndTensors` is in `jit/serialization/import.cpp`. This file is not included in mobile runtime compilation. That might be the reason CI `pytorch_macos_10_13_py3_lite_interpreter_build_test` fails.

iseeyuan commented on torch/csrc/jit/mobile/import.cpp:
```diff
@@ -459,25 +459,6 @@ std::unordered_map<std::string, std::string> BytecodeDeserializer::
 c10::IValue BytecodeDeserializer::readArchive(
     const std::string& archive_name,
     std::shared_ptr<mobile::CompilationUnit> mcu) {
-  std::stringstream picklename;
```
I love less duplicated codes!

iseeyuan reviewed (commented):
Nice to have more shared codes and less duplications. 

Could you please address the lite interpreter CI failure? I think it's caused by not including `serialization/import.cpp`. We cannot directly include `serialization/import.cpp` because it belongs to full-jit. Consider pulling `readArchiveAndTensors` out of `serialization/import.cpp`, and put it into a proper file that is shared by mobile and full-jit.

Could you also add comments in the summary, to show that this refactoring is not only to reduce duplications between mobile and server, but also to have a util function in cpp that could read .pkl files directly instead of loading the entire module.

cccclai committed:
Update on "[PyTorch][Edge] Reuse readArchiveAndTensors in mobile import"

## Summary 

The function from jit `torch::jit::readArchiveAndTensors` is quite general and can be reused in `mobile/import.h`. 

## Test plan
CI




[ghstack-poisoned]

cccclai commented:
Closing this pr, and have a different pr #57098 for this work. The comment is also directed to the new pr.
